cibyl/sources/elasticsearch/client.py

```diff
@@ -37,19 +37,36 @@ def __init__(self: object,
         :param port: Port for sending REST requests to the instance
         :type port: str, optional"""
         self.address = f"{host}:{port}"
+        self.connection = None
 
-    def connect(self: object) -> Elasticsearch:
+    def connect(self) -> Elasticsearch:
         """Connects to the elasticsearch instance
 
         :return: Elasticsearch module instance
         :rtype: Elasticsearch
         :raises: ElasticSearchError:
                  If exists an unhandled connection error
         """
-        es_client = Elasticsearch(self.address)
-        if not es_client.ping():
-            raise ElasticSearchError(f"Error connecting to "
-                                     f"Elasticsearch: {self.address}")
+        self.connection = Elasticsearch(self.address)
+        if not self.connection.ping():
+            message = 'Error connecting to '
+            message += f"Elasticsearch: {self.address}"
+            raise ElasticSearchError(message)
         LOG.debug(f"Connection established successfully with elasticsearch"
                   f" instance: {self.address}")
-        return es_client
+        return self
```

Comment:
Type hint points to ElasticSearch, which 'es_client' is, but you are returning 'self', which is a ElasticSearchClient.

Reply:
Changed. Thx.